lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -492,3 +568,84 @@ def disable(self) -> OpenSearchPluginConfig:
                 "s3.client.default.secret_key",
             ],
         )
+
+
+class OpenSearchGCSBackupPlugin(OpenSearchPlugin):
+    """Manage backup configurations for GCS.
+
+    This class must load the opensearch plugin: repository-gcs and configure it.
+    """
+
+    @property
+    def name(self) -> str:
+        """Returns the name of the plugin."""
+        return "repository-gcs"
+
+    def config(self) -> OpenSearchPluginConfig:
+        """Returns OpenSearchPluginConfig composed of configs used at plugin addition.
+
+        Format:
+        OpenSearchPluginConfig(
+            config_entries_to_add = {...},
+            config_entries_to_del = [...]|{...},
+            secret_entries_to_add = {...},
+            secret_entries_to_del = [...]|{...},
+        )
+        """
+        if not self._extra_config.get("service-account") and not self._extra_config.get(
+            "secret-key"
+        ):
+            raise OpenSearchPluginMissingConfigError(
+                "Missing GCS service-account or secret-key configuration",
+                affected_configs=["service-account", "secret-key"],
+                only_log=True,
+            )
+        import base64
+
+        data, conf = (
+            (self._extra_config["service-account"], "service-account")
+            if self._extra_config.get("service-account")
+            else (self._extra_config.get("secret-key"), "secret-key")
+        )
+        try:
+            account = " ".join(base64.b64decode(data).decode().rstrip().splitlines())
+        except Exception as e:
+            # This error means that GCS is wrongly configured.
```

Comment:
what kind of exact error are we expecting here, seems like `data` can never be null, so a base64 encoding will always succeed

Reply:
not if base64 is not power of 2 length OR not encoded at all.